### PR TITLE
Fix `islice` regression

### LINKED ISSUE #979: `islice` not stopping after end
Consider these examples:
```js
import { islice } from 'itertools';

function* positiveWholeNumbers() {
  let n = 1;
  while (true) {
    yield n++;
  }
}

const first5 = islice(positiveWholeNumbers(), 5);
console.log(Array.from(first5));
```

Expected behaviour: [1, 2, 3, 4, 5]
Current behaviour: Infinite loop

### PR DESCRIPTION
**Pull request by nvie**
Fixes #979.

### CODE REVIEW

File: src/itertools.ts [hunk 1]
```diff
@@ -201,9 +178,14 @@ export function* islice<T>(
...
-  const pred = slicePredicate(start, stop, step);
+  if (start < 0) throw new Error("start cannot be negative");
+  if (stop !== null && stop < 0) throw new Error("stop cannot be negative");
+  if (step <= 0) throw new Error("step cannot be negative");
+
   for (const [i, value] of enumerate(iterable)) {
-    if (pred(i)) {
+    if (i < start) continue;
+    if (stop !== null && i >= stop) break;
+    if (step === 1 || (i - start) % step === 0) {
```
Comment (Kareem-Medhat) [inline src/itertools.ts:188]: step === 1 is redundant as `n % 1` is always 0

File: src/itertools.ts [hunk 2]
```diff
@@ -201,9 +178,14 @@ export function* islice<T>(
...
       yield value;
```
Comment (Kareem-Medhat) [inline src/itertools.ts:189]: can also inverse the condition to keep same code style as before with early continues/breaks
```js
if ((i - start) % step !== 0) continue;
yield value;
```

File: src/itertools.ts (hunk 1 above)
Comment (nvie) [reply src/itertools.ts]: True, fixed in fe1f32c0caf06939222c9b2c18b42bcbdef6f751.

File: src/itertools.ts (hunk 2 above)
Comment (nvie) [reply src/itertools.ts:189]: I'm keeping it as-is, as to me the continue/break above select the start/end of the range, and this if-statement controls which values are skipped/emitted, which reads a bit clearer when using a positive condition.

**Comment by nvie** [comment]
Thanks for the pointer and the help here, @Kareem-Medhat 🙏 !

**Comment by nvie** [comment]
Published the fix in [v2.2.1](https://github.com/nvie/itertools/releases/tag/v2.2.1)!

**Pull request merged**